projects/output-and-error-handling/client/main.go

```diff
@@ -0,0 +1,182 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+)
+
+// Comments ----------
+// I write a lot of comments, it has helped me the entire course to learn, understand and remember
+// They can all be removed at the end to clean it up
+
+// Error Messages ----------
+// The Errors are deliberately numbered [x] so I could follow them up the call stack
+// They can be replaced with "user friendly" messages later
+
+func parseResponseBody(response *http.Response) (string, error) {
+	// check if there is a response body
+	if response.Body == nil {
```

Comment:
What is this check guarding against? Have you seen a case in real life where this code was needed? (I'm curious why you don't also feel the need to guard against `response` itself being `nil`, but do for `response.Body` which I assume should always be set by an `http.Client`?)

Reply:
i am not familiar enough with how a response is "created" (behind the scenes)